src/main/kotlin/org/koitharu/kotatsu/parsers/site/mangadventure/MangAdventureParser.kt

```diff
@@ -0,0 +1,208 @@
+
+	override val availableStates: Set<MangaState> = EnumSet.of(
+		MangaState.ONGOING,
+		MangaState.FINISHED,
+		MangaState.ABANDONED,
+		MangaState.PAUSED
+	)
+
+	override val availableContentRating: Set<ContentRating> =
+		EnumSet.of(ContentRating.SAFE)
+
+	override val availableSortOrders: Set<SortOrder> = EnumSet.of(
+		SortOrder.ALPHABETICAL,
+		SortOrder.ALPHABETICAL_DESC,
+		SortOrder.UPDATED,
+		SortOrder.POPULARITY
+	)
+
+	override val defaultSortOrder = SortOrder.ALPHABETICAL
+
+	override val isTagsExclusionSupported = true
+
+	override suspend fun getListPage(page: Int, filter: MangaListFilter?): List<Manga> {
+		val url = apiUrl.addEncodedPathSegment("series")
+			.addQueryParameter("limit", pageSize.toString())
+			.addQueryParameter("page", page.toString())
+		when (filter) {
+			is MangaListFilter.Search -> {
+				url.addQueryParameter("title", filter.query)
+			}
+			is MangaListFilter.Advanced -> {
+				url.addEncodedQueryParameter(
+					"categories",
+					buildString {
+						if (filter.tags.isNotEmpty() && filter.tagsExclude.isNotEmpty()) {
+							filter.tags.joinTo(this, ",", postfix = ",") { it.key.urlEncoded() }
+							filter.tagsExclude.joinTo(this, ",") { "-" + it.key.urlEncoded() }
+						} else if (filter.tags.isNotEmpty()) {
+							filter.tags.joinTo(this, ",") { it.key.urlEncoded() }
+						} else if (filter.tagsExclude.isNotEmpty()) {
+							filter.tagsExclude.joinTo(this, ",") { "-" + it.key.urlEncoded() }
+						}
+					}
+				)
+				when (filter.states.oneOrThrowIfMany()) {
+					null -> url.addEncodedQueryParameter("status", "any")
+					MangaState.ONGOING -> url.addEncodedQueryParameter("status", "ongoing")
+					MangaState.FINISHED -> url.addEncodedQueryParameter("status", "completed")
+					MangaState.ABANDONED -> url.addEncodedQueryParameter("status", "canceled")
+					MangaState.PAUSED -> url.addEncodedQueryParameter("status", "hiatus")
+					else -> throw IllegalArgumentException(ERROR_UNSUPPORTED_STATE)
+				}
+				when (filter.sortOrder) {
+					SortOrder.ALPHABETICAL -> url.addEncodedQueryParameter("sort", "title")
+					SortOrder.ALPHABETICAL_DESC -> url.addEncodedQueryParameter("sort", "-title")
+					SortOrder.UPDATED -> url.addEncodedQueryParameter("sort", "-latest_upload")
+					SortOrder.POPULARITY -> url.addEncodedQueryParameter("sort", "-views")
+					else -> throw IllegalArgumentException(ERROR_UNSUPPORTED_SORT_ORDER)
+				}
+			}
+			else -> {}
+		}
+		return runCatchingCancellable { getManga(url.get()) }.getOrElse {
+			if (it is NotFoundException) emptyList() else throw it
+		}
+	}
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val url = apiUrl.addEncodedPathSegment("series")
+			.addEncodedPathSegment(manga.slug)
+		val details = requireNotNull(url.get())
+		val chapters = url.addEncodedPathSegment("chapters")
+			.addEncodedQueryParameter("date_format", "timestamp").get()
+		return manga.copy(
+			description = details.optString("description", null),
```

Comment:
```suggestion
			description = details.getStringOrNull("description"),
```

I recommend to use extensions from JsonExt file instead of opt* functions because it may have a compatibility issues on Android (Android framework has a bit another org.json implementation). In addition to this extensions like `getLongOrDefault` and other are less strict to the data types that might be useful in case of non-strongly typed api